src/transformers/trainer_callback.py

```diff
@@ -475,3 +492,30 @@ def on_log(self, args, state, control, logs=None, **kwargs):
         _ = logs.pop("total_flos", None)
         if state.is_local_process_zero:
             print(logs)
+
+
+class EarlyStoppingCallback(TrainerCallback):
+    """
+    An early stopping callback
+    """
+
+    def check_metric_value(self, args, state, control, metric_value):
+        # best_metric is set by code for load_best_model
+        operator = np.greater if args.greater_is_better else np.less
+        if state.best_metric is None or (
+            operator(metric_value, state.best_metric)
+            and abs(metric_value - state.best_metric) > args.early_stopping_threshold
+        ):
+            state.early_stopping_patience_counter = 0
+        else:
+            state.early_stopping_patience_counter += 1
+
+    def on_best_metric_check(self, args, state, control, metric_value, **kwargs):
+        self.check_metric_value(args, state, control, metric_value)
+        print(state.early_stopping_patience_counter)
+        print(args.early_stopping_patience)
```

Comment:
Careful, remove your debugging prints ;-)

Reply:
Ope, darnit! I always think I get them all :laughing:

Reply:
Fixed in https://github.com/huggingface/transformers/pull/8581/commits/e75acaa0dcd51a955ca05695fcfb133ed07df2ab